src/nb_ndarray.cpp

```diff
@@ -335,12 +335,37 @@ ndarray_handle *ndarray_import(PyObject *o, const ndarray_req *req,
                 // c-style strides assumed
                 pass_order = req->req_order == 'C';
             } else {
-                for (uint32_t i = 0; i < (uint32_t) t.ndim; ++i) {
-                    if (!((strides[i] == t.strides[i]) ||
-                          (t.shape[i] == 1 && t.strides[i] == 0))) {
+                bool seen_previous_shape_eq_0 = false;
+                uint32_t i;
```

Comment:
This code is quite ugly. One nice thing would be to use a C++ iterator instead of lower-level indices. Just initialize the iterator to begin() if req_order = 'C', or rbegin() if req_order = 'F'. Also initialize end_it to either end() or rend().

Then we could drop the craziness going on at the bottom of the loop:

```
                    if ((req->req_order == 'C' && i == 0) ||
                        (req->req_order == 'F' && i == (uint32_t) (t.ndim - 1)))
                        break;
                    if (req->req_order == 'C') {
                        --i;
                    } else if (req->req_order == 'F') {
                        ++i;
                    }
```

and replace it with:

```
if (it == end_it) {
    break;
}
++it;
```